### PR TITLE
getResults: adding assertions

### PR DESCRIPTION
**Pull request by grewered**

### CODE REVIEW

**Review by weirichs — APPROVED** [review]
Thanks, LGTM

File: R/getResults.r [hunk 1]
```diff
@@ -1,7 +1,48 @@
...
+                       theta_init=NULL, print_iter = 20, verbose = TRUE, calc_ic=TRUE,
+                       omitUntil=1, seed=NA) {
+
+### checks ---------------------------------------------------------------------
+  checkmate::assert_list(runModelObj)
+
   q3MinType<- match.arg(q3MinType)
-  q3theta  <- match.arg(q3theta )
+  q3theta  <- match.arg(q3theta)
+  checkmate::assert_numeric(q3MinObs, len = 1)
```
Comment (weirichs) [inline R/getResults.r:20]: is something like `assert_integer` possible here? q3MinObs must be a single positive integer value.

File: R/getResults.r [hunk 2]
```diff
@@ -1,7 +1,48 @@
...
+         checkmate::assert_logical, len = 1)
+  #lapply(c(), checkmate::assert_logical, len = 1)
+
+  # if DIF was applied
+  lapply(c(abs.dif.bound, sig.dif.bound, p.value), checkmate::assert_numeric, len = 1)
+
+  # if software = "tam"
+  lapply(c(normal.approx, samp.regr, theta.model, beta_groups, sample_integers,
+           verbose, calc_ic), checkmate::assert_logical, len = 1)
+  checkmate::assert_numeric(nplausible, len = 1, null.ok = TRUE)
```
Comment (weirichs) [inline R/getResults.r:33]: `assert_integer`? nplausible must be a single positive integer value.

Comment (beckerbenj) [reply R/getResults.r:33]:
``` r
checkmate::assert_integer(1)
#> Error in eval(expr, envir, enclos): Assertion on '1' failed: Must be of type 'integer', not 'double'.
```

<sup>Created on 2024-10-24 with [reprex v2.0.2](https://reprex.tidyverse.org)</sup>

Comment (weirichs) [reply R/getResults.r:33]: Ah, thanks! `assert_integer` isn't a good idea. Then it can stay as it is.

**Pull request merged**